tests/driver_ws281x/Makefile

```diff
@@ -5,9 +5,8 @@ include ../Makefile.tests_common
 PIN ?= GPIO_PIN(0, 0)
 N ?= 8
 
-USEMODULE += ws281x
-
-FEATURES_REQUIRED := arch_avr8 # Currently only a backend for AVR available
+USEMODULE += ws281x # <-- Main platform independent driver
+USEMODULE += ws281x_atmega # <-- Backend used to communicate with the LEDs
```

Comment:
Can we resolve this in `Makefile.dep`?

Reply:
Like this?

Reply:
I was more thinking of

```Makefile
ifneq (,$(filter ws281x,$(USEMODULE)))
  ifneq (,$(filter arch_avr8,$(FEATURES_PROVIDED)))
    USEMODULE += ws281x_atmega
  else
    $(error No ws281x backend availiable for the current platform)
  endif
endif
```

Reply:
Travis says that modules must never inspect `FEASTURES_PROVIDED`. I dropped this to comply

Reply:
Does it also complain when you put this into `drivers/Makefile.dep`?

Reply:
That is exactly where I placed it before. As I didn't squash yet, you can still see it here: https://github.com/RIOT-OS/RIOT/pull/12693/commits/0c55316f8893022e8939d08ead7e890e483f27c0

(You could run `make static-test` with that to confirm locally.)

Reply:
Hm that is silly.
How about `ifneq (,$(filter atmega_common,$(USEMODULE)))`?

Reply:
This was also my first idea. But then I figured this is actually the same, but less elegant and readable. And being honest, it would be only to weasel around the check.

I'm not sure what the check wants to prevent. Maybe it wants to catch cases were `FEATURES_REQUIRED` should have been used, but the contributor was unaware and added a manual check of `FEATURES_PROVIDED` instead?

But until I know the actual reason, I thought it is better to back off. If my guess is correct, that message would be a false positive in the given case and the check could be improved. Or maybe there is a good reason why my first approach isn't a good idea that I'm just not aware of.

Reply:
When reading up on #10179 it seems like the original intention was that modules should rely on `FEATURES_USED` instead of `FEATURES_PROVIDED`, because just because a feature (e.g. SPI) is available doesn't mean that it's being used.

Adding the arch to `FEATURES_PROVIDED` happened later, but the arch is never 'used'.

So using `ifneq (,$(filter atmega_common,$(USEMODULE)))` would not be against the spirit of that check, but it would make using the driver much more user friendly.

I want to add an SDL backend for `native` so creating animations becomes easier to test - I just want to change the `BOARD` for that, not the `Makefile`.

Reply:
How about this?

Reply:
Yes this will work!